app/views/building_map/index.html.erb

```diff
@@ -43,29 +43,55 @@
 
 <datalist id="startOptions">
   <option><%= BuildingMapController::YOUR_LOCATION_MAGIC_STRING %></option>
-  <%= Places::DESTINATIONS.each do |dest, coords| %>
+  <% Places::DESTINATIONS.each do |dest, coords| %>
     <option>
       <%= dest %>
     </option>
   <% end %>
 </datalist>
 <datalist id="destOptions">
-  <%= Places::DESTINATIONS.each do |dest, coords| %>
+  <% Places::DESTINATIONS.each do |dest, coords| %>
     <option>
       <%= dest %>
     </option>
   <% end %>
 </datalist>
 
 <div class="container-fluid">
-  <%= map(
-        center: BuildingMapHelper.leaflet_center(@start),
-        polygons: BuildingMapHelper.leaflet_polygons,
-        polylines: BuildingMapHelper.leaflet_polylines(@route),
-        markers: BuildingMapHelper.leaflet_markers(@route, @target))
-  %>
+  <div id="map"></div>
 </div>
 
+<script>
```

Comment:
I would expect from ruby on rails that JS goes into some sort of `javascript` folder and should only be referenced here? (like views, controllers, stylesheets)

Reply:
Seems plausible

Reply:
Maybe the following could help: https://dev.to/morinoko/adding-custom-javascript-in-rails-6-1ke6 ?

Reply:
@chrisma 
We currently "parse" values from Ruby to JS like this: 
```erb
<script>
const view = <%= raw BuildingMapHelper.leaflet_center(@start).to_json %>;
</script>
```

Is it possible to pass values to a JS-script included via `javascript_pack_tag`? Or is interpolating the ruby values directly into an inline script the recommended way of doing this in ruby on rails?

Reply:
@bewee and @TheGrayStone advised us to make additional AJAX calls to get the data instead.

@chrisma is this an anti-pattern for ruby on rails? Are there general guidelines about how much and what should be done on the client-side and what on the server-side? I'm a bit afraid that we are ending up creating our own SPA if we do too much on the client...

Reply:
> Are there general guidelines about how much and what should be done on the client-side and what on the server-side?

Definitely makes sense to think about this. Concerning Ruby on Rails I would argue that the frameworks offers more support and power on the server-side (through models and controllers that can Ruby code) than it does on the "client" side (e.g. through erb templates). However, Ruby on Rails also works well to return JSON to the client side if this is required for certain features. I would probably recommend using the solution that is of low complexity and is easier to maintain in the future.